sdk/keyvault/keyvault-secrets/package.json

```diff
@@ -82,17 +82,17 @@
   },
   "dependencies": {
     "@azure/abort-controller": "^1.0.0",
-    "@azure/core-http": "^1.0.0",
+    "@azure/core-http": "^1.1.0",
     "@azure/core-lro": "^1.0.0",
     "@azure/core-paging": "^1.0.0",
-    "@azure/core-tracing": "1.0.0-preview.7",
+    "@azure/core-tracing": "1.0.0-preview.8",
     "@azure/logger": "^1.0.0",
-    "@opentelemetry/types": "^0.2.0",
```

Comment:
With the changes to secrets in this PR, we will now have 4.0.1 of certificates depending on a non released version of secrets which doesnt feel right. Can we not release secrets first, and then update the package.json file for certificates to depend on the released version of secrets?

Reply:
This secret dependency is just for tests, it's a devDependency. We can for sure do what you say, though it will make no difference for our users.

Reply:
It is less about the users and more about the precedence we are setting around maintaining our code. 

In the master branch, we have packages depending on other packages which are not yet released as well. But, this is fine as each of them are on their way to be released in the next applicable release cycle. This is indicated by the version number in the package.json of these packages which is greater than the released version.

In this hotfix branch, version 4.0.3 of secrets is being updated in place to make the tests of certificates work. Anyone looking at the 4.0.1 tag for keyvault-certificates in the future can misunderstand that 4.0.3 of secrets has the latest tracing support, when in reality, it is version 4.0.4 of secrets that will have the latest tracing support.

Reply:
Since we are being forced to change the package version of `core-tracing` to preview.7 in https://github.com/Azure/azure-sdk-for-js/pull/8872/commits/2804e58f747a2eef6b133c8e95684e1588e669b1?file-filters%5B%5D=.json to make rush download the right version, I guess my concern on package version and the code inside the package not matching is a ship that has sailed. :(

Reply:
It's a very odd scenario. keyvault-certificates in this branch is using an unpublished keyvault-secrets 4.0.3, as you say, which as you point out, doesn't align with the published 4.0.3 hotfix that will go based on the other hotfix branch.

To detect this though, users would need to clone the repository and go back to the hash of the hotfix release of keyvault-certificates, in which they will have a local copy of keyvault-secrets that would appear `(Unreleased)` in the changelog.

If they read certificates from their NPM bundle, they will only see the `^4.0.0` dependency.

I don't think there's any rush in releasing this hotfix before the keys and secrets hotfix, so we can pretty much wait and, as you suggested, release keys and secrets first, then update the certificate hotfix branches.

An easy way to do this update can be to force rush to download the 4.0.3 hotfix by pinning down keys and secrets like I'm doing with core-tracing: https://github.com/Azure/azure-sdk-for-js/pull/8872/commits/2804e58f747a2eef6b133c8e95684e1588e669b1 though this "pin down" approach mostly makes sense for core-tracing since we want to avoid having to update all of core-tracing dependencies by not pulling the actually published 1.0.0-preview.8.

In short, up to you! We can even delay this decision and focus on releasing the hotfix for keys and secrets.

Reply:
Keep in mind that I am aware that I'm mostly repeating what you're saying, but I want to make sure I communicate what I understand instead of assuming things.